students/morgan/dnd_project/encounter_builder/data_models/monster_data.py

```diff
@@ -0,0 +1,146 @@
+            cr = float(cls.cr)
+        except ValueError:
+            cr = 0
+
+
+        if 0 <= cr <= 1:
+            cls.size = sizes[random.randrange(0, 1+1)]
+            this_mon_focus = stat_focus[1]
+        elif 1 < cr <= 5:
+            cls.size = sizes[random.randrange(1, 3+1 )]
+            this_mon_focus = stat_focus[random.randrange(0,1+1)]
+        elif 6 <= cr <= 10:
+            cls.size = sizes[random.randrange(2, 4+1)]
+            this_mon_focus = stat_focus[random.randrange(0, 1 + 1)]
+        elif 11 <= cr:
+            cls.size = sizes[random.randrange(2, 5+1)]
+            this_mon_focus = stat_focus[0]
+
+        random.randint(0,10)
+
+        # print('str before',cls.str)
+        # print(this_mon_focus)
+        # print(cls.attack_bonus)
+
+        # print('focus', this_mon_focus)
+
+        if this_mon_focus == 'str':
+            # print('attack',cls.attack_bonus,'prof', cls.proficiency)
+            cls.str = math.floor((cls.attack_bonus - cls.proficiency)*2 )
+            cls.dex = random.randint(math.floor(-5 + (cr/random.randint(3,10))), cls.str)
+            cls.con = random.randint(math.floor(-5 + (cr/random.randint(3,10))), cls.str)
+            cls.wis = random.randint(math.floor(-5 + (cr/random.randint(3,10))), cls.str)
+            cls.int = random.randint(math.floor(-5 + (cr/random.randint(3,10))), cls.str)
+            cls.cha = random.randint(math.floor(-5 + (cr/random.randint(3,10))), cls.str)
+        elif this_mon_focus == 'dex':
+            # print('attack',cls.attack_bonus,'prof', cls.proficiency)
+
+            cls.dex = math.floor((cls.attack_bonus - cls.proficiency)*2 )
+            cls.str = random.randint(math.floor(-5 + (cr/random.randint(3,10))), cls.dex)
+            cls.con = random.randint(math.floor(-5 + (cr/random.randint(3,10))), cls.dex)
+            cls.wis = random.randint(math.floor(-5 + (cr/random.randint(3,10))), cls.dex)
+            cls.int = random.randint(math.floor(-5 + (cr/random.randint(3,10))), cls.dex)
+            cls.cha = random.randint(math.floor(-5 + (cr/random.randint(3,10))), cls.dex)
+
+        # print(cls.str, cls.dex)
+
+        if cr < 3:
+            cls.multi_attack = 1
+            cls.dmg_per_attack = cls.damage_rnd
+        else:
+            cls.multi_attack = math.ceil(cr/6)
+            cls.dmg_per_attack = cls.damage_rnd / cls.multi_attack
+
+class Enemy_Force():
+    def __init__(self, group_size, enemy_types, cr_range_easy, cr_range_medium, cr_range_hard, cr_range_daedly):
+        self.group_size = group_size
+        self.enemy_types = enemy_types
+        self.cr_range_easy = cr_range_easy
+        self.cr_range_medium = cr_range_medium
+        self.cr_range_hard = cr_range_hard
+        self.cr_range_deadly = cr_range_daedly
+        self.final_force = []
+        print('size', group_size)
+        print('types', enemy_types)
+
+    def build_force(self, cr_stats, difficulty):
+        # print(cr_stats)
+        if difficulty   == 1:
```

Comment:
this could be done with a  list or dict:

self.build_mon(stat_options=cr_stats, ranges[difficulty])

then you need  a list like:

self.ranges = [cr_range_easy, cr_range_medium, ...]